nan.h

```diff
@@ -1589,6 +1589,13 @@ class Callback {
   char *errmsg_;
 };
 
+// temaplate default is mainly set to have as little friction to prior
+// implementation of AsyncProgressWorker. The problem arises when the user needs
+// to send non-uniform types like the previous cons declaration in
+// HandleProgressCallback. Though this attempts backwards compatibility, cpp
+// does not allow for overfloaded pure virtual methods and conditional const or
+// de-const-ing.
+template<class T = char>
 /* abstract */ class AsyncProgressWorker : public AsyncWorker {
```

Comment:
I'd add a `struct AsyncProgressWorker : public AsyncProgressWorker<char> {};` for the sake of compatibility.